### ISSUE TITLE
Not able to run Discovery Service

### ISSUE THREAD

**Issue by buchan123** [odpi/egeria#4895]
Hello,
I am exploring Egeria project. I went through details like
	1. Egeria tutorial
	2. Egeria Dojo
	3. Egeria Labs
	I am able to basic things like creating Asset, adding more details to Asset.
I am following Egeria Labs.
I was trying to run a engine host server with a discovery engine inside it running a CSVDiscoveryService as its configured and shown in Egeria Dojo Labs.
I'm able to create and run the server,engine and CSV Discovery service.
Next step is,  I have created a CSV asset using Postman using the below endpoint.
Rest Endpoint : {{baseURL}}/servers/{{server}}/open-metadata/access-services/asset-owner/users/{{user}}/assets/data-files/csv
Body : {
			"class"       : "NewCSVFileAssetRequestBody",
			"displayName" : "My CSV File",
			"description" : "This is a test CSV file description added with Postman.",
			"fullPath"    : "home/my-table.csv"
		}
The rest endpoints provided in the DiscoveryMetadataStoreResource file inside discovery-engine-spring folder are not working for me even a simple searching of asset by-name isn't working.
Rest Endpoint : /servers/{serverName}/open-metadata/access-services/discovery-engine/users/{userId}/assets/by-name?startFrom=0&pageSize=10
Body : home/my-table.csv
I was checking the access-service Discovery Engine OMAS is in Technical Preview (considering it is developed but not released)
1. Can you please point out if I am following correct process for Discovery Engine /Discovery service ?
2. Please suggest if I should wait for exploring Discovery Engine for full release ?

**Comment by mandy-chessell** [comment]
What error are you getting from the call to the rest endpoint?

**Comment by buchan123** [comment]
when I'm hitting the rest endpoint this is the error message I'm getting in my Postman Client. @mandy-chessell 

{
    "class": "GUIDListResponse",
    "relatedHTTPCode": 400,
    "exceptionClassName": "org.odpi.openmetadata.frameworks.connectors.ffdc.PropertyServerException",
    "exceptionCausedBy": "java.lang.NullPointerException",
    "actionDescription": "getAssetsByName",
    "exceptionErrorMessage": "OMAG-COMMON-400-016 An unexpected java.lang.NullPointerException exception was caught by getAssetsByName; error message was null",
    "exceptionErrorMessageId": "OMAG-COMMON-400-016",
    "exceptionErrorMessageParameters": [
        "java.lang.NullPointerException",
        "getAssetsByName",
        "null"
    ],
    "exceptionSystemAction": "The system is unable to process the request and has returned an exception to the caller.",
    "exceptionUserAction": "Review the error message.  Also look up its full message definition which includes the system action and user action.  This is most likely to describe the correct action to take to resolve the error.  If that does not help, look for other diagnostics created at the same time.  Also validate that the caller is a valid client of this server and is operating correctly."
}

**Comment by mandy-chessell** [comment]
Apologies for the slow response - we are still in the middle of refactoring the discovery support from the discovery server to the engine host server.  The work is close to completion.  However, I did look specifcally at the API calls that you reported as failing and it was bug introduced by the refactoring work.  I have fixed them so thank you for reporting the problems.  I hope to have finished the discovery refactoring in the next few weeks.
